src/sipTransportConstructor.ts

```diff
@@ -0,0 +1,176 @@
+import { WebPhoneUserAgent } from './userAgent';
+import { Timers, Transport } from 'sip.js';
+
+
+export interface WebPhoneSIPTransport extends Transport {    
+    computeRandomTimeout: typeof computeRandomTimeout;
+    reconnect: typeof reconnect;
+    isSipErrorCode: typeof isSipErrorCode;
+    scheduleSwithBackMainProxy: typeof scheduleSwithBackMainProxy;
+    onSipErrorCode: typeof onSipErrorCode;
+}
+
+
+export const TransportConstructorWrapper = (SipTransport: any, webPhoneOptions:any): any => {
+    
+    return function transportConstructor(logger: Function, options:any): WebPhoneSIPTransport {
+
+        let transport = new SipTransport(logger, options);
+
+        transport.nextReconnectInterval = 0;
+        transport.sipErrorCodes = webPhoneOptions.sipErrorCodes;
+        transport.switchBackInterval = webPhoneOptions.switchBackInterval;
+
+        transport.computeRandomTimeout = computeRandomTimeout.bind(transport);
+        transport.reconnect = reconnect.bind(transport);
+        transport.isSipErrorCode = isSipErrorCode.bind(transport);
+        transport.scheduleSwithBackMainProxy = scheduleSwithBackMainProxy.bind(transport);
+        transport.onSipErrorCode = onSipErrorCode.bind(transport);
+        transport.__isCurrentMainProxy = __isCurrentMainProxy.bind(transport);
+        transport.__afterWSConnected = __afterWSConnected.bind(transport);
+        transport.__onConnectedToBackup = __onConnectedToBackup.bind(transport);
+        transport.__onConnectedToMain = __onConnectedToMain.bind(transport);
+
+        transport.on('connected', transport.__afterWSConnected);
+
+        return transport;
+    }
+}
+
+const C = {
+                // Transport status codes
+                STATUS_CONNECTING: 0,
+                STATUS_OPEN: 1,
+                STATUS_CLOSING: 2,
+                STATUS_CLOSED: 3
+    };
+
+function computeRandomTimeout(reconnectionAttempts: number, randomMinInterval: number, randomMaxInterval: number){
+            randomMinInterval = randomMinInterval < 0 ? 0 : randomMinInterval;
+            randomMaxInterval = randomMaxInterval <= 0 ? 7000 : randomMaxInterval;
+            reconnectionAttempts = reconnectionAttempts || 1;

```

Comment:
This can be moved to definition of the function:
```ts
function computeRandomTimeout(reconnectionAttempts: number, randomMinInterval: number, randomMaxInterval: number = 1)
```